openmeter/entitlement/adapter/entitlement.go

```diff
@@ -166,6 +172,39 @@ func (a *entitlementDBAdapter) DeleteEntitlement(ctx context.Context, entitlemen
 	return err
 }
 
+func (a *entitlementDBAdapter) DeactivateEntitlement(ctx context.Context, entitlementID models.NamespacedID, at time.Time) error {
+	_, err := entutils.TransactingRepo[interface{}, *entitlementDBAdapter](
+		ctx,
+		a,
+		func(ctx context.Context, repo *entitlementDBAdapter) (interface{}, error) {
+			ent, err := repo.GetEntitlement(ctx, entitlementID)
+			if err != nil {
+				return nil, err
+			}
+
+			if ent.ActiveTo != nil {
+				return nil, fmt.Errorf("entitlement %s is already deactivated", entitlementID.ID)
+			}
+
+			c, err := repo.db.Entitlement.Update().
```

Comment:
Why not using `Entitlement.UpdateOneID()` instead?

Reply:
So i can throw an error if it's already deactivated

Reply:
Sorry, i really miscomprehended that